tests/solana_tests/builtin.rs

```diff
@@ -101,17 +105,21 @@ fn pda() {
                 return create_program_address([a, b], program_id);
             }
 
-            function create_pda2_bump() public returns (address, bytes1) {
+            function create_pda2_bump(bool cond) public returns (address, bytes1) {
```

Comment:
What do you want to achieve with this boolean here?
All your tests set this argument as `true`.

Reply:
This is to ensure that `create_program_address()` is not in the first block of the cfg; that is the problem case.

Reply:
I think we should still be testing `create_program_address` as the first instruction in a block.
What about this?

```solidity
function create_pda(bool cond) public returns (address) {
                 address program_id = address"BPFLoaderUpgradeab1e11111111111111111111111";
                 address addr =  create_program_address(["Talking", "Cats"], program_id);
                 if (cond) {
                     return create_program_address(["Talking", "Squirrels"], program_id);
                 } else {
                     return addr;
                 }
             }
```

Reply:
I've added this test